### PR TITLE
enhance: return the auth collection list when using the `show collection` api

### PR DESCRIPTION
**Pull request by SimFG**
issue: #32550
/kind improvement

### CODE REVIEW

File: internal/rootcoord/show_collection_task.go [hunk 1]
```diff
@@ -45,6 +47,76 @@ func (t *showCollectionTask) Prepare(ctx context.Context) error {
...
+
+	getVisibleCollections := func() (typeutil.Set[string], error) {
+		enableAuth := Params.CommonCfg.AuthorizationEnabled.GetAsBool()
+		privilegeColls := typeutil.NewSet[string]()
+		if !enableAuth {
+			privilegeColls.Insert(util.AnyWord)
+			return privilegeColls, nil
+		}
+		curUser, err := contextutil.GetCurUserFromContext(ctx)
+		if err != nil || curUser == util.UserRoot {
```
Comment (czs007) [inline internal/rootcoord/show_collection_task.go:59]: Under what circumstances would errors occur ?

Comment (czs007) [reply internal/rootcoord/show_collection_task.go:59]: add a log.Warn ?

**Comment by SimFG** [comment]
rerun ut

**Pull request merged**